src/daft-distributed/src/pipeline_node/mod.rs

```diff
@@ -49,13 +54,21 @@ impl RunningPipelineNode {
     pub fn into_inner(self) -> Receiver<PipelineOutput> {
         self.result_receiver
     }
+
+    #[allow(dead_code)]
+    pub fn materialize(self, task_dispatcher_handle: TaskDispatcherHandle) {
+        materialize_all_pipeline_outputs(
+            ReceiverStream::new(self.result_receiver).map(Ok),
+            task_dispatcher_handle,
+        );
+    }
 }
 
 impl Stream for RunningPipelineNode {
```

Comment:
let's making this into a seperate struct. The code should look like 
```
let running_pipeline_node = ...
let stream = running_pipeline_node.into_stream();
...
```

Reply:
You can just implement this trait for that so it works with other adapters

https://docs.rs/futures/latest/futures/stream/struct.IntoStream.html